lib/Onyx.js

```diff
@@ -1571,7 +1587,25 @@ function update(data) {
         }
     });
 
-    return clearPromise.then(() => Promise.all(_.map(promises, (p) => p())));
+    return clearPromise.then(() => Storage.multiGet(_.keys(updateQueue)).then((pairs) => {
+        const existingSnapshot = _.reduce(pairs, (acc, [key, value]) => {
+            acc[key] = value;
+            return acc;
+        }, {});
+
+        const updatedSnapshot = {};
+        _.each(updateQueue, (operations, key) => {
+            let existingValue = existingSnapshot[key];
+            if (operations[0].onyxMethod === METHOD.SET) {
+                existingValue = operations[0].value;
+            }
+
+            const mergeOperations = _.filter(operations, ({onyxMethod}) => onyxMethod === METHOD.MERGE);
```

Comment:
Since a single operation can't be both a `SET` and a `MERGE` there might be a little cleaning up to this logic which could happen (maybe an early return in the `SET` case?)

Reply:
The logic behind this is the following: I'm putting the `merge` and `set` operations in an array of a certain object key in the order they appear in the update operation.
However, since the `set` operation fully overwrites the existing object, it makes sense to keep only the operations that come after the last `set` (if it is present).

Therefore, eventually, a single key's update queue will consist of an array of `merge` operations with (optionally) only the first one in the array being `set`: `[set?, merge, merge, ...]`.

Does it make sense?

Reply:
Ah, OK. That makes sense. Adding some code comments could help with that.

Does this also work in a case like this where a single key might have multiple operations that are separated by other keys?

1. key1 - merge
2. key2 - merge
3. key1 - merge

That should result in a single merge instruction for `key1`.

Reply:
> Does this also work in a case like this where a single key might have multiple operations that are separated by other keys?
> 
> 1. key1 - merge
> 2. key2 - merge
> 3. key1 - merge
> 
> That should result in a single merge instruction for `key1`.

Yes, I'll perform some tests and attach different scenarios in a while